netlogo-gui/src/main/app/NotificationBanner.scala

```diff
@@ -0,0 +1,251 @@
+import java.util.prefs.{Preferences => JPreferences}
+import org.nlogo.core.I18N
+
+import java.awt.event.{FocusEvent, FocusListener}
+import scala.io.Source
+import scala.collection.immutable.List
+
+case class JsonObject(eventId: Int, date: String, title: String, fullText: String)
+
+class NotificationBanner() extends JPanel with ThemeSync {
+  private var jsonObjectList: List[JsonObject] = List() // Initialize here first
+  private val JsonUrl = "https://ccl.northwestern.edu/netlogo/announce-test.json"
+  // Fetch and populate jsonObjectList in the constructor
+  jsonObjectList = parseJsonToList(fetchJsonFromUrl(JsonUrl))
+
+  private val initialMessage = getJsonObjectHead.getOrElse("")
+  lazy val lastSeenEventIdKey: String = "lastSeenEventId" // The key for the most recently seen event-id
+
+  // Label to display notification messages
+  private val messageLabel = new JLabel(" " + initialMessage)
+
+  // Close button with an "X" icon
+  private val closeButton = new JButton("\u2716") // Unicode character for "X"
+  setVisible(isShowNeeded()) // Set visibility based on isShowNeeded
+  // Initialize the NotificationBanner panel
+  initUI()
+
+  def syncTheme(): Unit = {
+    setBackground(InterfaceColors.MONITOR_BACKGROUND)
+    messageLabel.setForeground(InterfaceColors.DISPLAY_AREA_TEXT)
+    closeButton.setForeground(InterfaceColors.DISPLAY_AREA_TEXT)
+  }
+
+  // Method to fetch JSON content from a URL
+  def fetchJsonFromUrl(url: String): String = {
+    try {
+      val source = Source.fromURL(url)
+      val content = source.mkString
+      source.close()
+      content
+    } catch {
+      case e: Exception =>
+      println(s"Unable to connect to ccl.northwestern.edu but not throwing an exception")
+      ""
+//        throw new Exception(s"Error fetching JSON from URL: ${e.getMessage}", e)
+    }
+  }
+
+  private def initUI(): Unit = {
+    // Set layout and appearance
+    setLayout(new BorderLayout())
+    setPreferredSize(new Dimension(400, 50))
+
+    // Customize the message label
+    messageLabel.setHorizontalAlignment(SwingConstants.LEFT)
+    add(messageLabel, BorderLayout.CENTER)
+
+    // Configure the close button
+    closeButton.setBorderPainted(false)
+    closeButton.setContentAreaFilled(false)
+    closeButton.setFocusPainted(false)
+    //TODO this action needs to get called on the okay button click.
+    closeButton.addActionListener(new ActionListener {
+      override def actionPerformed(e: ActionEvent): Unit = {
+        setVisible(false) // Hide the banner when the close button is pressed
+      }
+    })
+    add(closeButton, BorderLayout.EAST)
+
+    // Add a mouse listener to call showJsonInDialog when the banner is clicked
+    addMouseListener(new MouseAdapter {
+      override def mouseClicked(e: MouseEvent): Unit = {
+        showJsonInDialog() // Call the method on click
+      }
+    })
+  }
+
+  // Method to update the notification message
+  def setMessage(message: String): Unit = {
```

Comment:
This function is never used, it can be removed.

Reply:
It's now removed